content/general/concepts/algorithm/terms/counting-sort/counting-sort.md

```diff
@@ -0,0 +1,121 @@
+---
+Title: 'Counting Sort'
+Description: 'Sorts a list with duplicate values efficiently.'
+Subjects:
+  - 'Computer Science'
+  - 'Interview Prep'
+Tags:
+  - 'Algorithms'
+  - 'Sorting Algorithms'
+  - 'Java'
+  - 'Counting Sort'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**Counting sort** is an out-of-place, non-comparison sorting [algorithm](https://www.codecademy.com/resources/docs/general/algorithm) that sorts a list with duplicate values efficiently. It is a helpful algorithm for sorting [arrays](https://www.codecademy.com/resources/docs/java/arrays) of small, non-negative integers, where it analyzes elements' frequency distribution and places them in their final sorted positions.
+
+## Explanation
+
+1. Find the maximum value:
+
+- Iterate through the input array `data` to find the largest element `max`.
+  - This determines the range of values we need to count.
```

Comment:
Avoid using first- and second-person pronouns (e.g. "I", "we", "you") if possible. Please refactor this.

Reply:
just removed the first person pronouns, thanks for the heads up